sdk/core/core-util/test/public/datetime.spec.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { toOffsetDateTime } from "../../src/datetime";
+import { assert } from "chai";
+
+describe("utcDateTime with timezone offset", function () {
```

Comment:
Could we add more cases to cover corner situations e.g +0, -12?

Reply:
As stated here https://github.com/Azure/azure-sdk-for-js/pull/27087#discussion_r1322330577

Reply:
Let me know if you can figure out a way.